Top100/top100songs.py

```diff
@@ -0,0 +1,257 @@
+
+    # Handle multiple Artists
+    if "With" in artist:
+        artist = artist.split(' With')[0]
+    if 'Featuring' in artist:
+        artist = artist.split(' Featuring')[0]
+    if '&' in artist:
+        artist = artist.split(' & ')[0]
+
+    # Handle bracketed song title synonyms
+    if '(' in song_title:
+        song_title = song_title.split(' (')[0]
+
+    # Create Artist page URL address
+    base = 'https://www.lyrics.com/'
+    url = base + 'artist/' + artist.lower().replace(' ', '+')
+    relative_path = ''
+
+    content = get_content_from_url(url)
+    songs = content.find_all('td', {'class': 'tal qx'})
+
+    # Non-specific artist name return empty string, page is redirected to a search artist page
+    if len(songs) == 0:
+        # Look-Up in all Relevant artist
+        artists = content.find_all('td', {'class': 'tal fx'})
+
+        # Search in all matching artists for the song
+        for artist_link in artists:
+            try:
+                url = base + artist_link.a.attrs['href']
+            except AttributeError:
+                continue
+            content = get_content_from_url(url)
+            songs = content.find_all('td', {'class': 'tal qx'})
+            relative_path = find_song_title_link(song_title, songs)
+            if relative_path:
+                break
+    else:
+        relative_path = find_song_title_link(song_title, songs)
+
+    return base + relative_path
+
+
+def get_lyrics(song_title: str, artist: str) -> Union[str, None]:
+    """The get_lyrics function gets the lyrics for a song_title,
+    from https://www.lyrics.com website
+    and return all the lyrics
+
+    :param song_title: str.
+    :param artist: str
+    :return: lyrics: str. None if no URL was found.
+    """
+    url = get_full_url_to_lyrics(song_title, artist)
+    if url is None:
+        return ''
+
+    lyr_page = get_content_from_url(url)
+    lyrics = lyr_page.find('pre', {'id': 'lyric-body-text'})  # , limit=1)
+    return lyrics.text
+
+
+def count_words_in_text(text: str) -> dict:
+    """Counts the number of unique words in a given string.
+    :param text: str
+    :return: word_count: dict
+    """
+    # Check input validation
+    if text is '':
+        return dict()
+    if type(text) is not str:
+        raise ValueError("Input must be string")
+
+    # Clean rows from Text
+    text.replace("\n", ' ')
+
+    # Insert word without punctuations
+    word_count = {}
+    for word in text.split():
+        word = word.strip(string.punctuation).lower()
+        try:
+            word_count[word] = word_count[word] + 1
+        except KeyError:
+            word_count[word] = 1
+    return word_count
+
+
+def get_top_100_songs() -> dict:
+    """This function gets the top 100 songs as published on the
+    Billboard hot-100 page (https://www.billboard.com/charts/hot-100)
+    :return: songs_dict: dict
+    """
+    url = 'https://www.billboard.com/charts/hot-100'
```

Comment:
URLs are also magics. Upper case it and pick a better name.

Reply:
renamed it yet the uppercase thing is just raising problems in pycharm ide